packages/tradershub/src/hooks/useCurrencies.ts

```diff
@@ -14,34 +16,68 @@ export type TCurrencies = {
 
 /** A custom hook to get the currency config information from `website_status` endpoint and in predefined order */
 const useCurrencies = () => {
-    const { data, ...rest } = useQuery('website_status');
+    const { data: authorizeData, isLoading: isAuthorizeLoading } = useAuthorize();
+    const { data: websiteStatusData, isLoading: isWesiteStatusLoading, ...rest } = useQuery('website_status');
+    const { data: landingCompanyData, isLoading: isLandingCompanyLoading } = useLandingCompany();
+    const { isNonEU } = useRegulationFlags();
 
+    // Get the legal allowed currencies based on the landing company
+    const legalAllowedCurrencies = useMemo(() => {
+        if (!landingCompanyData) return [];
+        if (isNonEU) {
```

Comment:
the flag coming from `useRegulationFlags` is enough to handle the filtering out of currencies for CR, MF & DIEL clients